### PR TITLE
[#43692] Part 1. Modify API to be able to handle file_links without a container.

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/43692

The PR adds a new endpoint `POST /api/v3/file_links` and a cron job to delete hung file_links.

### CODE REVIEW

File: modules/storages/lib/api/v3/file_links/file_link_representer.rb [hunk 1]
```diff
@@ -144,7 +145,7 @@ def _type
     private
 
     def user_allowed_to_manage?(model)
-      current_user.allowed_to?(:manage_file_links, model.container.project)
+      model.container && current_user.allowed_to?(:manage_file_links, model.project)
```
Comment (Kharonus) [inline modules/storages/lib/api/v3/file_links/file_link_representer.rb]: minor: can you make that `nil` check here explicit? With 

```ruby
model.container.present?
```

File: modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb [hunk 2]
```diff
@@ -91,6 +91,76 @@
...
+          end
+          unset_keys.each do |key|
+            expect(file_link.attributes[key]).to be_nil
+          end
+        end
+      end
+
+      it 'does not provide a link to the collection of created file links' do
+        self_link = JSON.parse(response.body).dig('_links', 'self', 'href')
+        expect(self_link).to eq('urn:openproject-org:api:v3:file_links:no_link_provided')
```
Comment (Kharonus) [inline modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb]: Use capybara matcher for json pathes:

```ruby
expect(response.body).to be_json_eql('urn:openproject-org:api:v3:file_links:no_link_provided').at_path('_links/self/href')
```

Comment (Kharonus) [reply modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb]: same below

File: modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb [hunk 3]
```diff
@@ -91,6 +91,76 @@
     login_as current_user
   end
 
+  describe 'POST /api/v3/file_links' do
+    let(:path) { '/api/v3/file_links' }
+    let(:permissions) { %i(manage_file_links) }
+    let(:storage_url1) { storage.host }
+    let(:storage_url2) { storage.host }
```
Comment (Kharonus) [inline modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb]: No need to use 2 times the same storage host in different variables, or do I miss something?

File: modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb [hunk 4]
```diff
@@ -91,6 +91,76 @@
...
+
+      it 'creates corresponding FileLink records', :aggregate_failures do
+        expect(Storages::FileLink.count).to eq 2
+        Storages::FileLink.find_each.with_index do |file_link, i|
+          unset_keys = %w[container_id container_type]
+          set_keys = (file_link.attributes.keys - unset_keys)
+          set_keys.each do |key|
+            expect(file_link.attributes[key]).not_to(
+              be_nil,
+              "expected keyibute #{key.inspect} of FileLink ##{i + 1} to be set.\ngot nil."
```
Comment (Kharonus) [inline modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb]: what is a keyibute?

**Review by Kharonus — CHANGES_REQUESTED** [review]
Only some minor issues.

File: modules/storages/lib/api/v3/file_links/file_link_representer.rb (hunk 1 above)
Comment (ba1ash) [reply modules/storages/lib/api/v3/file_links/file_link_representer.rb]: Yes, sure.

File: modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb (hunk 3 above)
Comment (ba1ash) [reply modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb]: Im not sure. I will check it.

File: modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb (hunk 4 above)
Comment (ba1ash) [reply modules/storages/spec/requests/api/v3/file_links/file_links_spec.rb]: It is an orphan of a substitute command run on a region. I will fix it.

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**